pkg/controller/elasticsearch/user/roles.go

```diff
@@ -18,12 +18,46 @@ const (
 	SuperUserBuiltinRole = "superuser"
 	// ProbeUserRole is the name of the role used by the internal probe user.
 	ProbeUserRole = "elastic_internal_probe_user"
+
+	// ApmUserRoleV6 is the name of the role used by the association user for 6.8.x APMServer instances.
+	ApmUserRoleV6 = "eck_apm_user_role_v6"
+	// ApmUserRoleV7 is the name of the role used by the association user for APMServer instances from version 7.1 to 7.4 included.
+	ApmUserRoleV7 = "eck_apm_user_role_v7"
+	// ApmUserRoleV75 is the name of the role used by the association user for APMServer instances from version 7.5
+	ApmUserRoleV75 = "eck_apm_user_role_v75"
 )
 
 var (
 	// PredefinedRoles to create for internal needs.
 	PredefinedRoles = RolesFileContent{
 		ProbeUserRole: esclient.Role{Cluster: []string{"monitor"}},
+		ApmUserRoleV6: esclient.Role{
+			Cluster: []string{"monitor", "manage_index_templates"},
+			Indices: []esclient.IndexRole{
+				{
+					Names:      []string{"apm-*"},
+					Privileges: []string{"write", "create_index"},
+				},
+			},
+		},
+		ApmUserRoleV7: esclient.Role{
+			Cluster: []string{"monitor", "manage_ilm", "manage_index_templates"},
+			Indices: []esclient.IndexRole{
+				{
+					Names:      []string{"apm-*"},
+					Privileges: []string{"manage", "write", "create_index"},
```

Comment:
`"manage"` is actually needed only from 7.3 but I'm not sure it's worth it to introduce a specific role for 7.1 and 7.2

Reply:
I'd keep it simple if it doesn't break anything for 7.1 and 7.2